### PR TITLE
feat: change challenge desc to ts

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: client/src/templates/Challenges/components/Challenge-Description.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { Fragment } from 'react';
+
+import PrismFormatted from './PrismFormatted';
+import './challenge-description.css';
+
+type Challenges = {
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/Challenge-Description.tsx:6]:
```suggestion
type Challenge = {
```

File: client/src/templates/Challenges/components/Challenge-Description.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  block: string;
+  description: string;
+  instructions: string;
+};
+
+function ChallengeDescription(challenge: Challenges) {
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/Challenge-Description.tsx:12]:
```suggestion
function ChallengeDescription(challenge: Challenge) {
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Most pedantic of nitpicks aside, this LGTM.

**Comment by Sembauke** [comment]
I also see that there is a Missing return type on function

**Comment by ShaunSHamilton** [comment]
> What should the return type be? JSX.Element ?

Usually, yes.

**Comment by Sembauke** [comment]
> FYI: I disabled auto-merge to remind me to go and edit the migration issue when we merge.

I already did that for you :)

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Ah, so you did.  Thanks.

I'll check it off once this goes in.

File: client/src/templates/Challenges/components/Challenge-Description.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React, { Fragment } from 'react';
+
+import PrismFormatted from './PrismFormatted';
+import './challenge-description.css';
+
+type Challenge = {
+  block: string;
+  description: string;
+  instructions: string;
+};
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/components/Challenge-Description.tsx:10]: At some point we'll likely want to build this out as a full exportable type, independent of component files.

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

File: client/src/templates/Challenges/components/Challenge-Description.tsx (hunk 3 above)
Comment (Sembauke) [reply client/src/templates/Challenges/components/Challenge-Description.tsx:10]: I will keep that in mind

**Pull request merged**